File: java/DistributedTypeValidation.java
// Example 2: DistributedTypeValidation.java

class OneToTen {
    int ott;
    OneToTen(int ott) {
        this.ott = ott;
    }
    @Override
    public String toString() {
        return "OneToTen(" + ott + ")";
    }
}

public class DistributedTypeValidation {
    static OneToTen f1(OneToTen x) {
        Check.validity(0 < x.ott && x.ott <= 10, x.ott + " out of range");
        return new OneToTen(x.ott * 10);
    }
    static OneToTen f2(OneToTen x) {
        Check.validity(0 < x.ott && x.ott <= 10, x.ott + " out of range");
        return new OneToTen(x.ott + 10);
    }
    public static void main(String[] args) {
        var a = new OneToTen(6);
        System.out.println(a);
        System.out.println(f1(a));
        System.out.println(f2(a));
        var b = new OneToTen(11);
        System.out.println(f1(b));
        a.ott = 99;
        System.out.println(f2(a));
    }
}

File: kotlin/DistributedTypeValidation.kt
// Example 2: DistributedTypeValidation.java

data class OneToTen(var ott: Int) {
    override fun toString(): String {
        return "OneToTen($ott)"
    }
}

fun f1(x: OneToTen): OneToTen {
    check(0 < x.ott && x.ott <= 10, x.ott.toString() + " out of range")
    return OneToTen(x.ott * 10)
}

fun f2(x: OneToTen): OneToTen {
    check(0 < x.ott && x.ott <= 10, x.ott.toString() + " out of range")
    return OneToTen(x.ott + 10)
}

fun main() {
    val a = OneToTen(6)
    System.out.println(a)
    System.out.println(f1(a))
    System.out.println(f2(a))
    val b = OneToTen(11)
    System.out.println(f1(b))
    a.ott = 99
    System.out.println(f2(a))
}
